tests/utils.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
+
+import logging
+import sys
+from datetime import datetime
+from typing import Any
+
+from airflow.configuration import secrets_backend_list
+from airflow.exceptions import AirflowSkipException
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.models.taskinstance import TaskInstance
+from airflow.secrets.local_filesystem import LocalFilesystemBackend
+from airflow.utils import timezone
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.utils.state import DagRunState, State
+from airflow.utils.types import DagRunType
+from sqlalchemy.orm.session import Session
+
+log = logging.getLogger(__name__)
+
+
+def run_dag(dag: DAG, conn_file_path: str) -> DagRun:
+    return test_dag(dag=dag, conn_file_path=conn_file_path)
+
+
+@provide_session
+def test_dag(
```

Comment:
Please add a note mentioning that we have to copy DAG.test for older airflow versions